File: app/src/main/java/com/example/finalproject/api/RecipesApi.java
package com.example.finalproject.api;

import com.example.finalproject.model.MusicalsResult;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Query;

public interface RecipesApi {

    @Headers("Api-Key: 1234")
    @GET("recipes/v2?type=public&q=chicken&app_id=546a4f67&app_key" +
            "=aba8a9a207fd136545e0d95803581aa5&imageSize=SMALL")
    Call<MusicalsResult> getAllRecipes();
}

File: app/src/main/java/com/example/finalproject/api/RecipesApi.kt
package com.example.finalproject.api

import com.example.finalproject.model.RecipeResponse
import com.google.gson.GsonBuilder
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Query
 interface RecipesApi {
    @GET("api/recipes/v2/?type=public")
     suspend fun getAllRecipes(
        @Query("q") query: String,
        @Query("app_id") appId: String,
        @Query("app_key") appKey: String,
        @Query("imageSize") imageSize: String
    ):  Response<RecipeResponse>

     companion object {
         operator fun invoke(): RecipesApi {
             val BASE_URL = "https://api.edamam.com/"
             val gson = GsonBuilder()
                 .setLenient()
                 .create()

             return Retrofit.Builder()
                 .baseUrl(BASE_URL)
                 .addConverterFactory(GsonConverterFactory.create(gson))
                 .build()
                 .create(RecipesApi::class.java)
         }
     }
}
